dbos/client.go

```diff
@@ -162,28 +162,32 @@ func (c *client) Enqueue(queueName, workflowName string, input any, opts ...Enqu
 	}
 
 	uncancellableCtx := WithoutCancel(dbosCtx)
-
-	tx, err := dbosCtx.systemDB.(*sysDB).pool.Begin(uncancellableCtx)
-	if err != nil {
-		return nil, newWorkflowExecutionError(workflowID, fmt.Sprintf("failed to begin transaction: %v", err))
-	}
-	defer tx.Rollback(uncancellableCtx) // Rollback if not committed
-
-	// Insert workflow status with transaction
-	insertInput := insertWorkflowStatusDBInput{
-		status: status,
-		tx:     tx,
-	}
-	_, err = dbosCtx.systemDB.insertWorkflowStatus(uncancellableCtx, insertInput)
+	err := retry(dbosCtx, func() error {
```

Comment:
Why do we need retries here? It's the client, it can just fail, there's no deeper risk

Reply:
Not necessarily strictly speaking, but nice to have from a reliable library imo. Right now, all the client methods, which are wrapping the package -> context methods, will retry.

Are you suggesting we should do the minimum amount of retries to avoid getting a corrupted Transact process (e.g., queue runner unusable), but should let the user handle connection errors on all other paths? E.g., cancel workflow.

Reply:
I like the idea of doing less. We can easily add retries to client methods (enqueue, but also RetrieveWorkflow, CancelWorkflow, ResumeWorkflow, ListWorkflows and ForkWorkflow) later if need be. The most important is to avoid having unusable-but-running Transact processes.